kernel/interrupt.c

```diff
@@ -98,14 +101,17 @@ static void user_irq_queue_delete(int irq)
 
 static inline void user_irq_reset_all(void)
 {
-	for (int i = 0 ; i < IRQn_NUM ; i++)
+	int i;
+	for (i = 0 ; i < IRQn_NUM ; i++)
```

Comment:
Can't your compiler support C99 features like the statement `for (int i = 0 ; i < IRQn_NUM ; i++)`?